bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/callmonitor/CallEvent.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.avmfritz.internal.callmonitor;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Call Events received from a fritzbox.
+ *
+ * 12.07.20 09:11:30;RING;0;0171123456;888888;SIP2;
+ * 12.07.20 09:13:40;DISCONNECT;0;0;
+ *
+ * @author Kai Kreuzer - Initial contribution
+ */
+@NonNullByDefault
+public class CallEvent {
+
+    private final String rawEvent;
+    private final String timestamp;
+    private final String callType;
+    private final String id;
+    private final String externalNo;
+    private final @Nullable String internalNo;
+    private final @Nullable String connectionType;
+    private final @Nullable String line;
+
+    public CallEvent(String rawEvent) {
+        this.rawEvent = rawEvent;
+
+        String[] fields = rawEvent.split(";");
+        if (fields.length < 4) {
+            throw new IllegalArgumentException("Cannot parse call event: " + rawEvent);
+        }
+
+        timestamp = fields[0];
+        callType = fields[1];
+        id = fields[2];
+
+        if (callType.equals("RING")) {
```

Comment:
I would change the order to "RING".equals(callType) to avoid NPEs later on

Reply:
callType is NonNull, so there is no risk.